### ISSUE TITLE
SEV-SNP: snp-host-latest kernel causes large slowdown during KVM_CREATE_VM

### ISSUE THREAD

**Issue by bencw12** [AMDESE/AMDSEV#145]
I recently updated my host kernel to the 6.1.0 kernel on the [snp-host-latest](https://github.com/AMDESE/linux/tree/snp-host-latest) branch and I noticed a ~50ms slowdown (although it seems to vary between ~30ms and ~70ms) when executing KVM_CREATE_VM for all guest VMs regardless of SEV. I've tracked the delay down to the creation of a worker thread for NX huge page recovery as seen below:

From arch/x86/kvm/mmu/mmu.c
```
int kvm_mmu_post_init_vm(struct kvm *kvm)
{
        int err;

        pr_info("BCWH kvm_vm_create_worker_thread: START");
        err = kvm_vm_create_worker_thread(kvm, kvm_nx_huge_page_recovery_worker, 0,
                                          "kvm-nx-lpage-recovery",
                                          &kvm->arch.nx_huge_page_recovery_thread);
        pr_info("BCWH kvm_vm_create_worker_thread: DONE");

        if (!err)
                kthread_unpark(kvm->arch.nx_huge_page_recovery_thread);

        return err;
}
```

```
$ sudo dmesg
...
[   41.681135] kvm: BCWH kvm_vm_create_worker_thread: START
[   41.737124] kvm: BCWH kvm_vm_create_worker_thread: DONE
...
```

In the example above creating the worker thread adds ~56ms of additional overhead. I've been testing in Firecracker with non-SEV guests where the guest boots in 50-60ms with my old 5.15 kernel so this is adding a large delay that I'm hoping to get to the bottom of. Is this slowdown intrinsic to the kernel support for SNP? If so, could it be avoided for non-SEV guests? Any and all help is much appreciated!

**Comment by tlendacky** [comment]
I don't think it is anything specific to SEV. Have you tried a base 6.1.0 kernel to see if the same thing occurs?

**Comment by bencw12** [comment]
Yes, I actually tried with a clean 5.19, 6.1, and 6.3 kernel earlier today and found the same slowdown, so it must be something else. Sorry about that!
